sample_size/metrics.py

```diff
@@ -0,0 +1,65 @@
+from statsmodels.stats.power import NormalIndPower
+from statsmodels.stats.power import TTestIndPower
+
+
+class BooleanMetric:
+    def __init__(
+        self,
+        probability: float,
+        mde: float,
+    ):
+        self.probability = self._get_probability(probability)
+        self.variance = self._get_variance()
+        self.mde = mde
+        self.default_power_analysis_type = NormalIndPower
```

Comment:
I update the metrics to store an instance of the power solver. Would a base class be necessary here?

Reply:
I will +1 on adding a base class to connect these three classes for their common pattern, at least to define and make sure the common variable names (`self.variance`, `self.mde`, `self.default_power_analysis_type`) are consistent. Since they are used by calculator generally, with base class defined we can avoid mismatch variable names

Reply:
added